### PR TITLE
UI Changes for inverted theme, and supporting extensibility of CSS in custom.css

### PR DESCRIPTION
**Pull request by microsoftsam**
- Moving some CSS code to pxt.less (so that targets can overwrite it)
- Monaco editor error view zone changes to support inverted theme
- Moving some layout CSS into a pxt.less so that variables can be overwritten in a target to control the layout, eg: menu height, simulator width
- Minor UI fixes to support inverted theme 
- Removing unnecessary borders

### CODE REVIEW

File: theme/pxt.less [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  position: absolute;
+  top:@mainMenuHeight;
+  bottom: 0rem;
+}
+#sidedocs {
+  display:none;
+  overflow-y: auto;
+  right: 0;
+  border:none;
+  height: calc(100% - 4.5rem);
```
Comment (pelikhan) [inline theme/pxt.less]: Mainmenuheight?

File: theme/pxt.less [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    #maineditor {
+        top:@mobileMenuHeight !important;
+    }
+}
+
+
+/* extra-large desktop screen */
+@media only screen and (min-width: 92em) and (min-height:30em) {
+    .sideDocs #maineditor {
+        right:@sideBarWidth;
```
Comment (pelikhan) [inline theme/pxt.less]: Nice!

File: theme/pxt.less [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    }
+    .rtl .sideDocs #maineditor {
+        left:@sideBarWidth;
+        right:@simulatorWidth;
```
Comment (pelikhan) [inline theme/pxt.less]: Finally!

File: theme/pxt.less (hunk 1 above)
Comment (microsoftsam) [inline theme/pxt.less]: Menubar height.

**Pull request merged**